### PR TITLE
An enhancement when multiple HystrixConcurrencyStrategy detected. Fix…

### PR DESCRIPTION
**Pull request by ailinzhou**
Add multiple `HystrixConcurrencyStrategy` check. #3256

### CODE REVIEW

File: spring-cloud-netflix-hystrix/src/main/java/org/springframework/cloud/netflix/hystrix/security/HystrixSecurityAutoConfiguration.java [hunk 1]
```diff
@@ -56,18 +57,33 @@ public void init() {
...
 		HystrixPlugins.getInstance().registerCommandExecutionHook(commandExecutionHook);
 	}
 
+	private HystrixConcurrencyStrategy getConcurrencyStrategy() {
+		HystrixConcurrencyStrategy registeredStrategy = HystrixPlugins.getInstance()
+				.getConcurrencyStrategy();
+		//Hystrix Registered a default Strategy.
+		//if registeredStrategy not the default and not some with existingConcurrencyStrategy throw a exception.
+		if (registeredStrategy != null
+				&& !registeredStrategy.equals(HystrixConcurrencyStrategyDefault.getInstance())
```
Comment (ryanjbaxter) [inline spring-cloud-netflix-hystrix/src/main/java/org/springframework/cloud/netflix/hystrix/security/HystrixSecurityAutoConfiguration.java]: Could we just check if its an `instanceof`?

Comment (ailinzhou) [reply spring-cloud-netflix-hystrix/src/main/java/org/springframework/cloud/netflix/hystrix/security/HystrixSecurityAutoConfiguration.java]: Thanks for you good advice, it more clearer. I add test file and change  `HystrixSecurityAutoConfiguration#getConcurrencyStrategy` for easy to read and  understand. Hope to hear advice from you.

**Comment by ryanjbaxter** [comment]
Why two PRs? https://github.com/spring-cloud/spring-cloud-netflix/pull/3276/files

**Comment by ailinzhou** [comment]
> Why two PRs? #3276

It's my mistake. When I push some commit, I realized squash those commit to one commit will better for you review, so I do squash. But It seems that github already cached what I have squash. In order to remove squashed commits. I remove my forked repository. this made I can't push commit in this PR and you can see github shows tip `commits into spring-cloud:master from` **unknown repository**.
In this situation I create a new PR #3276 .
could you please help me close this PR and review PR #3276 @ryanjbaxter .

**Pull request closed without merging**